parlai/agents/seq2seq/modules.py

```diff
@@ -5,17 +5,34 @@
 # of patent rights can be found in the PATENTS file in the same directory.
 
 import math
+
 import torch
 import torch.nn as nn
 from torch.nn.parameter import Parameter
 from torch.nn.utils.rnn import pad_packed_sequence, pack_padded_sequence
 import torch.nn.functional as F
-from parlai.core.torch_agent import Beam
-from parlai.core.dict import DictionaryAgent
-import os
+
+
+def opt_to_kwargs(opt):
+    """Get kwargs for seq2seq from opt."""
+    kwargs = {}
+    for k in ['numlayers', 'dropout', 'bidirectional', 'rnn_class',
+              'lookuptable', 'decoder', 'numsoftmax',
+              'attention', 'attention_length', 'attention_time']:
+        if k in opt:
+            kwargs[k] = opt[k]
+    return kwargs
 
 
 def pad(tensor, length, dim=0):
+    """Pad tensor to a specific length.
+
+    :param tensor: vector to pad
+    :param length: new length
+    :param dim: (default 0) dimension to pad
+
+    :returns: padded tensor if the tensor is shorter than length
+    """
     if tensor.size(dim) < length:
         return torch.cat(
             [tensor, tensor.new(*tensor.size()[:dim],
```

Comment:
Just out of pure curiosity, did you benchmark this with other approaches? (namely init the full zero matrix, then assign the tensor into it?). Not that there's need to.